From f183b60fad2a1d264f8a16031179012785acebae Mon Sep 17 00:00:00 2001
From: Ranjani Vaidyanathan <ra5478@freescale.com>
Date: Tue, 13 Nov 2012 16:08:41 -0600
Subject: ENGR00233705 [MX6SL] -Fix suspend/resume issue when SD1 is used to
 boot.

Setting certain IOMUX settings on SD1 prevents the system from
entering suspend. These pins are already configured as GPIO, so
it does not help to reconfigure them during suspend.

Signed-off-by: Ranjani Vaidyanathan <ra5478@freescale.com>
---
 arch/arm/mach-mx6/board-mx6sl_common.h |    6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6sl_common.h b/arch/arm/mach-mx6/board-mx6sl_common.h
index a6e5e90..1170cb6 100644
--- a/arch/arm/mach-mx6/board-mx6sl_common.h
+++ b/arch/arm/mach-mx6/board-mx6sl_common.h
@@ -461,6 +461,7 @@ static iomux_v3_cfg_t suspend_enter_pads[] = {
 	MX6SL_PAD_SD1_DAT5__GPIO_5_9,
 	MX6SL_PAD_SD1_DAT6__GPIO_5_7,
 	MX6SL_PAD_SD1_DAT7__GPIO_5_10,
+
 	MX6SL_PAD_SD2_CLK__GPIO_5_5,
 	MX6SL_PAD_SD2_CMD__GPIO_5_4,
 	MX6SL_PAD_SD2_DAT0__GPIO_5_1,
@@ -469,18 +470,19 @@ static iomux_v3_cfg_t suspend_enter_pads[] = {
 	MX6SL_PAD_SD2_DAT3__GPIO_4_28,
 	MX6SL_PAD_SD2_DAT4__GPIO_5_2,
 	MX6SL_PAD_SD2_DAT5__GPIO_4_31,
-	MX6SL_PAD_SD2_DAT6__GPIO_4_29,
-	MX6SL_PAD_SD2_DAT7__GPIO_5_0,
+
 	MX6SL_PAD_SD3_CLK__GPIO_5_18,
 	MX6SL_PAD_SD3_CMD__GPIO_5_21,
 	MX6SL_PAD_SD3_DAT0__GPIO_5_19,
 	MX6SL_PAD_SD3_DAT1__GPIO_5_20,
 	MX6SL_PAD_SD3_DAT2__GPIO_5_16,
 	MX6SL_PAD_SD3_DAT3__GPIO_5_17,
+
 	/* USBOTG ID pin */
 	MX6SL_PAD_EPDC_PWRCOM__GPIO_2_11,
 	MX6SL_PAD_HSIC_STROBE__GPIO_3_20,
 	MX6SL_PAD_HSIC_DAT__GPIO_3_19,
+
 	/* Key row/column */
 	MX6SL_PAD_KEY_COL0__GPIO_3_24,
 	MX6SL_PAD_KEY_COL1__GPIO_3_26,
-- 
1.7.9.5

